Admin/AbstractAdmin.php

```diff
@@ -543,6 +544,13 @@
     private $breadcrumbsBuilder;
 
     /**
+     * The route id handler.
+     *
+     * @var RouteIdHandlerInterface
+     */
+    private $routeIdHandler;
```

Comment:
Do we need this here?

Reply:
I do not know =)

But...
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L277

Adds method call setRouteIdHandler(). 

And may be:
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Controller/CRUDController.php#L1409
```php
    protected function getObjectIdFromRouteIdHandler()
    {
        return $this->admin->getRouteIdHandler()->getIdFromRequest($this->getRequest(), $this->admin);
    }
```